src/openvic-simulation/economy/FactoryProducer.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <cstdint>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/pop/Pop.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+#include "utility/Constructors.hpp"
+
+namespace OpenVic {
+	class FactoryProducer final {
+	private:
+		static constexpr uint8_t DAYS_OF_HISTORY = 7;
+
+	public:
+		typedef ordered_map<Pop* const, Pop::pop_size_t> employees_map_t;
+		typedef std::array<fixed_point_t, DAYS_OF_HISTORY> daily_profit_history_t;
```

Comment:
I don't believe typedef is any less type-safe, its just a pain to manage because its ordered weird and isn't clear about what it refers to, typedef and using aliases are functionally synonyms.